### ISSUE TITLE
X-server fails with segmentation fault (dev-channel) on devices with no freon yet.

### ISSUE THREAD

**Issue by divx118** [dnschneid/crouton#1450]
Should be related to https://github.com/dnschneid/crouton/issues/1443

Update: This is an issue when using X-method xiwi or xorg.

```
(trusty)maurice@localhost:~$ croutonversion
crouton: version 1-20150226194717~master:841ff620
release: trusty
architecture: amd64
xmethod: xiwi
targets: xiwi,xfce,keyboard,extension,chrome
host: version 6812.3.0 (Official Build) dev-channel falco
```

/tmp/Xorg.crouton.1.log
```
(trusty)maurice@localhost:~$ cat /tmp/Xorg.crouton.1.log
[  4129.736]
X.Org X Server 1.15.1
Release Date: 2014-04-13
[  4129.736] X Protocol Version 11, Revision 0
[  4129.736] Build Operating System: Linux 3.2.0-76-generic x86_64 Ubuntu
[  4129.736] Current Operating System: Linux localhost 3.8.11 #1 SMP Tue Feb 24 17:59:04 PST 2015 x86_64
[  4129.736] Kernel command line: cros_secure console= loglevel=7 init=/sbin/init cros_secure oops=panic panic=-1 root=/dev/dm-0 rootwait ro dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 dm="1 vroot none ro 1,0 2506752 verity payload=PARTUUID=36d4e9de-a997-2042-b788-b5689686594f/PARTNROFF=1 hashtree=PARTUUID=36d4e9de-a997-2042-b788-b5689686594f/PARTNROFF=1 hashstart=2506752 alg=sha1 root_hexdigest=1e7082e200e2de94c72bcb478fde375374e76571 salt=0956530ddc7baee714805bf90b3029546e40254ff53725a84cb7d3d6b3dfaad0" noinitrd vt.global_cursor_default=0 kern_guid=36d4e9de-a997-2042-b788-b5689686594f add_efi_memmap boot=local noresume noswap i915.modeset=1 tpm_tis.force=1 tpm_tis.interrupts=0 nmi_watchdog=panic,lapic iTCO_vendor_support.vendorsupport=3  
[  4129.736] Build Date: 12 February 2015  02:49:29PM
[  4129.736] xorg-server 2:1.15.1-0ubuntu2.7 (For technical support please see http://www.ubuntu.com/support)
[  4129.736] Current version of pixman: 0.30.2
[  4129.736]    Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
[  4129.736] Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
[  4129.736] (++) Log file: "/tmp/Xorg.crouton.1.log", Time: Fri Feb 27 19:25:51 2015
[  4129.739] (++) Using config file: "/etc/X11/xorg-dummy.conf"
[  4129.739] (==) Using system config directory "/usr/share/X11/xorg.conf.d"
[  4129.743] (==) ServerLayout "Layout0"
[  4129.743] (**) |-->Screen "Screen0" (0)
[  4129.743] (**) |   |-->Monitor "Monitor0"
[  4129.744] (**) |   |-->Device "Card0"
[  4129.744] (**) Option "DontVTSwitch" "true"
[  4129.744] (**) Option "AllowMouseOpenFail" "true"
[  4129.744] (**) Option "AutoAddDevices" "false"
[  4129.744] (**) Option "AutoEnableDevices" "false"
[  4129.744] (**) Option "AutoAddGPU" "false"
[  4129.744] (**) Not automatically adding devices
[  4129.744] (**) Not automatically enabling devices
[  4129.744] (**) Not automatically adding GPU devices
[  4129.750] (WW) The directory "/usr/share/fonts/X11/cyrillic" does not exist.
[  4129.750]    Entry deleted from font path.
[  4129.750] (WW) The directory "/usr/share/fonts/X11/100dpi/" does not exist.
[  4129.750]    Entry deleted from font path.
[  4129.751] (WW) The directory "/usr/share/fonts/X11/75dpi/" does not exist.
[  4129.751]    Entry deleted from font path.
[  4129.751] (WW) The directory "/usr/share/fonts/X11/100dpi" does not exist.
[  4129.751]    Entry deleted from font path.
[  4129.751] (WW) The directory "/usr/share/fonts/X11/75dpi" does not exist.
[  4129.751]    Entry deleted from font path.
[  4129.751] (==) FontPath set to:
        /usr/share/fonts/X11/misc,
        /usr/share/fonts/X11/Type1,
        built-ins
[  4129.751] (==) ModulePath set to "/usr/lib/x86_64-linux-gnu/xorg/extra-modules,/usr/lib/xorg/extra-modules,/usr/lib/xorg/modules"
[  4129.751] (==) |-->Input Device "<default pointer>"
[  4129.751] (==) |-->Input Device "<default keyboard>"
[  4129.751] (==) The core pointer device wasn't specified explicitly in the layout.
        Using the default mouse configuration.
[  4129.751] (==) The core keyboard device wasn't specified explicitly in the layout.
        Using the default keyboard configuration.
[  4129.751] (II) Loader magic: 0x7f508b85cd40
[  4129.751] (II) Module ABI versions:
[  4129.751]    X.Org ANSI C Emulation: 0.4
[  4129.751]    X.Org Video Driver: 15.0
[  4129.751]    X.Org XInput driver : 20.0
[  4129.751]    X.Org Server Extension : 8.0
[  4129.751] (II) xfree86: Adding drm device (/dev/dri/card0)
[  4129.752] (II) xfree86: Adding drm device (/dev/dri/card1)
[  4129.752] (EE)
[  4129.752] (EE) Backtrace:
[  4129.752] (EE) 0: /usr/bin/Xorg (xorg_backtrace+0x48) [0x7f508b5d8848]
[  4129.752] (EE) 1: /usr/bin/Xorg (0x7f508b42f000+0x1ad539) [0x7f508b5dc539]
[  4129.752] (EE) 2: /lib/x86_64-linux-gnu/libpthread.so.0 (0x7f508a52c000+0x10340) [0x7f508a53c340]
[  4129.752] (EE) 3: /usr/bin/Xorg (0x7f508b42f000+0xb57a6) [0x7f508b4e47a6]
[  4129.753] (EE) 4: /usr/bin/Xorg (xf86BusProbe+0x9) [0x7f508b4b8099]
[  4129.753] (EE) 5: /usr/bin/Xorg (InitOutput+0x74d) [0x7f508b4c66fd]
[  4129.753] (EE) 6: /usr/bin/Xorg (0x7f508b42f000+0x59bab) [0x7f508b488bab]
[  4129.753] (EE) 7: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main+0xf5) [0x7f5088f6cec5]
[  4129.753] (EE) 8: /usr/bin/Xorg (0x7f508b42f000+0x451ee) [0x7f508b4741ee]
[  4129.753] (EE)
[  4129.753] (EE) Segmentation fault at address 0x0
[  4129.753] (EE)
Fatal server error:
[  4129.753] (EE) Caught signal 11 (Segmentation fault). Server aborting
[  4129.753] (EE)
[  4129.753] (EE)
Please consult the The X.Org Foundation support
         at http://wiki.x.org
 for help.
[  4129.753] (EE) Please also check the log file at "/tmp/Xorg.crouton.1.log" for additional information.
[  4129.753] (EE)
```

**Comment by dnschneid** [comment]
Looks like the hack I added needs to be split out and applied to Xorg regardless of freon.

**Comment by dnschneid** [comment]
Could you please try both xiwi and xorg on latest master and post the xorg.log if either is still broken?

**Comment by divx118** [comment]
Both work now again thanks.

**Comment by divx118** [comment]
hmm I thought it worked, it seems a bit of hit and miss. So strange...
```
chronos@localhost / $ sudo enter-chroot -n test

Entering /mnt/stateful_partition/crouton/chroots/test...
(test)maurice@localhost:~$ croutonversion
crouton: version 1-20150228231818~master:cafd3891
release: trusty
architecture: amd64
xmethod: xorg
targets: xfce
host: version 6812.3.0 (Official Build) dev-channel falco 

(test)maurice@localhost:~$ cat /var/log/Xorg.2.log

[ 19789.515] 
X.Org X Server 1.15.1
Release Date: 2014-04-13
[ 19789.515] X Protocol Version 11, Revision 0
[ 19789.515] Build Operating System: Linux 3.2.0-76-generic x86_64 Ubuntu
[ 19789.515] Current Operating System: Linux localhost 3.8.11 #1 SMP Tue Feb 24 17:59:04 PST 2015 x86_64
[ 19789.515] Kernel command line: cros_secure console= loglevel=7 init=/sbin/init cros_secure oops=panic panic=-1 root=/dev/dm-0 rootwait ro dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 dm="1 vroot none ro 1,0 2506752 verity payload=PARTUUID=36d4e9de-a997-2042-b788-b5689686594f/PARTNROFF=1 hashtree=PARTUUID=36d4e9de-a997-2042-b788-b5689686594f/PARTNROFF=1 hashstart=2506752 alg=sha1 root_hexdigest=1e7082e200e2de94c72bcb478fde375374e76571 salt=0956530ddc7baee714805bf90b3029546e40254ff53725a84cb7d3d6b3dfaad0" noinitrd vt.global_cursor_default=0 kern_guid=36d4e9de-a997-2042-b788-b5689686594f add_efi_memmap boot=local noresume noswap i915.modeset=1 tpm_tis.force=1 tpm_tis.interrupts=0 nmi_watchdog=panic,lapic iTCO_vendor_support.vendorsupport=3  
[ 19789.516] Build Date: 12 February 2015  02:49:29PM
[ 19789.516] xorg-server 2:1.15.1-0ubuntu2.7 (For technical support please see http://www.ubuntu.com/support) 
[ 19789.516] Current version of pixman: 0.30.2
[ 19789.516]    Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
[ 19789.516] Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
[ 19789.516] (==) Log file: "/var/log/Xorg.2.log", Time: Sun Mar  1 16:06:42 2015
[ 19789.517] (==) Using system config directory "/usr/share/X11/xorg.conf.d"
[ 19789.517] (==) No Layout section.  Using the first Screen section.
[ 19789.517] (==) No screen section available. Using defaults.
[ 19789.517] (**) |-->Screen "Default Screen Section" (0)
[ 19789.517] (**) |   |-->Monitor "<default monitor>"
[ 19789.517] (==) No device specified for screen "Default Screen Section".
        Using the first device section listed.
[ 19789.517] (**) |   |-->Device "Intel Graphics SNA+TearFree"
[ 19789.517] (==) No monitor specified for screen "Default Screen Section".
        Using a default monitor configuration.
[ 19789.517] (==) Automatically adding devices
[ 19789.517] (==) Automatically enabling devices
[ 19789.517] (==) Automatically adding GPU devices
[ 19789.517] (WW) The directory "/usr/share/fonts/X11/cyrillic" does not exist.
[ 19789.517]    Entry deleted from font path.
[ 19789.517] (WW) The directory "/usr/share/fonts/X11/100dpi/" does not exist.
[ 19789.517]    Entry deleted from font path.
[ 19789.518] (WW) The directory "/usr/share/fonts/X11/75dpi/" does not exist.
[ 19789.518]    Entry deleted from font path.
[ 19789.518] (WW) The directory "/usr/share/fonts/X11/100dpi" does not exist.
[ 19789.518]    Entry deleted from font path.
[ 19789.518] (WW) The directory "/usr/share/fonts/X11/75dpi" does not exist.
[ 19789.518]    Entry deleted from font path.
[ 19789.518] (==) FontPath set to:
        /usr/share/fonts/X11/misc,
        /usr/share/fonts/X11/Type1,
        built-ins
[ 19789.518] (==) ModulePath set to "/usr/lib/x86_64-linux-gnu/xorg/extra-modules,/usr/lib/xorg/extra-modules,/usr/lib/xorg/modules"
[ 19789.518] (II) The server relies on udev to provide the list of input devices.
        If no devices become available, reconfigure udev or disable AutoAddDevices.
[ 19789.518] (II) Loader magic: 0x7fb04e16ad40
[ 19789.518] (II) Module ABI versions:
[ 19789.518]    X.Org ANSI C Emulation: 0.4
[ 19789.518]    X.Org Video Driver: 15.0
[ 19789.518]    X.Org XInput driver : 20.0
[ 19789.518]    X.Org Server Extension : 8.0
[ 19789.518] (II) xfree86: Adding drm device (/dev/dri/card0)
[ 19789.518] (II) xfree86: Adding drm device (/dev/dri/card1)
[ 19789.519] (EE) 
[ 19789.519] (EE) Backtrace:
[ 19789.519] (EE) 0: /usr/bin/X (xorg_backtrace+0x48) [0x7fb04dee6848]
[ 19789.519] (EE) 1: /usr/bin/X (0x7fb04dd3d000+0x1ad539) [0x7fb04deea539]
[ 19789.519] (EE) 2: /lib/x86_64-linux-gnu/libpthread.so.0 (0x7fb04ce3a000+0x10340) [0x7fb04ce4a340]
[ 19789.520] (EE) 3: /usr/bin/X (0x7fb04dd3d000+0xb57a6) [0x7fb04ddf27a6]
[ 19789.520] (EE) 4: /usr/bin/X (xf86BusProbe+0x9) [0x7fb04ddc6099]
[ 19789.520] (EE) 5: /usr/bin/X (InitOutput+0x74d) [0x7fb04ddd46fd]
[ 19789.520] (EE) 6: /usr/bin/X (0x7fb04dd3d000+0x59bab) [0x7fb04dd96bab]
[ 19789.520] (EE) 7: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main+0xf5) [0x7fb04b87aec5]
[ 19789.520] (EE) 8: /usr/bin/X (0x7fb04dd3d000+0x451ee) [0x7fb04dd821ee]
[ 19789.521] (EE) 
[ 19789.521] (EE) Segmentation fault at address 0x0
[ 19789.521] (EE) 
Fatal server error:
[ 19789.521] (EE) Caught signal 11 (Segmentation fault). Server aborting
[ 19789.521] (EE) 
[ 19789.521] (EE) 
Please consult the The X.Org Foundation support 
         at http://wiki.x.org
 for help. 
[ 19789.521] (EE) Please also check the log file at "/var/log/Xorg.2.log" for additional information.
[ 19789.521] (EE)
```

**Comment by divx118** [comment]
Some more info: A precise chroot works without problems. Also xiwi works on trusty and precise. 
Maybe if of interest a good xorg log of starting a precise chroot. http://pastebin.com/Zabht7F0
So only XMETHOD xorg is not working on trusty and maybe debian chroots, but didn't try those yet.

**Comment by dnschneid** [comment]
Oh wow, even in the working one, it still tries to load card1 for some reason.  I'm amazed that even worked.

**Comment by dnschneid** [comment]
Can you post the outputs of `sudo strace -f xinit &> strace.log` for both working and non-working ones?

**Comment by divx118** [comment]
Well if running with strace I get a working X window with xterm on trusty so no segmentation fault. [strace.log](http://www.timara.nl/crouton/strace.log)
The Precise strace.log is [strace_precise.log](http://www.timara.nl/crouton/strace_precise.log)

BTW tried setting serverflag Option "AutoAddGPU" "false", that doesn't make a difference. In the strace logs everything both chroots are back to stock so no modifications. Both are chroots with xfce target only installed.

**Comment by dnschneid** [comment]
Whoops, pretty certain my fix isn't working simply because LD_PRELOAD doesn't work with setuid.  That's why it works with xiwi (non-root) and not xorg (root).

**Comment by dnschneid** [comment]
OK.  Try the gemfix branch and see if that fixes things.  This latest hack makes me sad.

**Comment by divx118** [comment]
For trusty patching xorg works, also X server starts with no problems.
However for precise it fails. I just copy paste the relevant piece when running the installer with -x. If more is needed just say so.
```
+ ln -sfT /etc/crouton/xserverrc /etc/X11/xinit/xserverrc
+ echo xorg
+ grep -F -m 1 -boa card[0-9]* /usr/bin/Xorg
+ offset=1930638:card[0-9]*
+ [ -n 1930638:card[0-9]* ]
+ dd seek=1930638 bs=1 of=/usr/bin/Xorg conv=notrunc,nocreat status=none
dd: invalid status flag: `none'
Try `dd --help' for more information.
+ echo -n croutonhax
+ set +e
+ trap - INT HUP TERM 0
```

**Comment by divx118** [comment]
Ok, with the latest push precise is also fixed.
However on precise and on trusty we have an unknown display added in croutoncycle. So switching is a bit borked, because mouse and keyboard will switch to the unknown and display will stay the same.
```
chronos@localhost / $ sudo enter-chroot -n precise croutoncycle list
Password: 
Entering /mnt/stateful_partition/crouton/chroots/precise...
6291457  Unknown
cros* Chrome OS
:1  precise
```
Trust xorg log http://pastebin.com/8sJEpwci
Precise xorg log http://pastebin.com/tGw7B8aT

**Comment by dnschneid** [comment]
Yay.  I think that's a separate problem.  Try unmounting all chroots, then doing `sudo enter-chroot croutonwebsocket` and see if the extension reports the unknown window.  It's probably the host-x11 window filtering code being tricked by something new in Chromium OS, which if so, deserves a separate bug filed.

**Comment by dnschneid** [comment]
OK, I've merged that fix into master.  @smibarber's fingers are resting on the "revert" button in case we start getting a bunch of reports coming in while I'm out on vacation.

**Comment by DennisLfromGA** [comment]
I knew I was a little concerned when you mentioned 'vacation' but you *really* got the `crouton-bot` all riled up ;)

Have a good one... :+1:

**Comment by divx118** [comment]
`sudo enter-chroot -n trusty croutonwebsocket` does only report the chromeos window, so that is good. I will create a new issue. It can be really annoying if you log out in the chroot, there is no easy way to get control of your system again. Keyboard, mouse doesn't switch back to chromeos, also the crouton extension clipboard sync doesn't work when running XMETHOD xorg.

**Comment by ravomendo** [comment]
it appears that latest release of Chrome Dev OS broke Crouton again.  Same issues.  It worked, and then I updated Chrome OS, and it broke again.  Similar issues reported here.

**Comment by divx118** [comment]
@ravomendo After the update did you update your chroot again? I had to do that, because my HP chromebook 14 made the switch from X server to freon on chromeos.

**Comment by ravomendo** [comment]
well, I tried to follow the update procedure without any success.  I would try again.

**Comment by errietta** [comment]
fwiw I had this same problem (trusty chroot) - was resolved by updating the chroot as per https://github.com/dnschneid/crouton/#a-new-version-of-crouton-came-out-my-chroot-is-therefore-obsolete-and-sad

**Comment by ameensol** [comment]
@errietta I just had the same problem (also trusty chroot). I'm trying to update the chroot as per the instructions in the readme, but am having two problems.

1) croutonversion doesn't do anything on my machine, so I can't check for updates that way. I ended up downloading the latest crouton via the [link](http://goo.gl/fd3zc) shared by @The-MAZZTer in his [issue](https://github.com/dnschneid/crouton/issues/788).  

2) When I try and update crouton via `sudo sh -e ~/Downloads/crouton -n trusty -u -t unity`, I get the following error:

```
Err http://ppa.launchpad.net trusty/main amd64 Packages            
  404  Not Found
Err http://ppa.launchpad.net trusty/main i386 Packages             
  404  Not Found
Ign http://ppa.launchpad.net trusty/main Translation-en            
Ign http://repo.steampowered.com precise/steam Translation-en      
W: Failed to fetch http://ppa.launchpad.net/pitti/postgresql/ubuntu/dists/trusty/main/binary-amd64/Packages  404  Not Found

W: Failed to fetch http://ppa.launchpad.net/pitti/postgresql/ubuntu/dists/trusty/main/binary-i386/Packages  404  Not Found

E: Some index files failed to download. They have been ignored, or old ones used instead.
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/trusty...
```

I don't think the two postgres files should be that important, but either way it prevents my chroot from updating and I get the same segfault error when I `sudo startunity`.

**Comment by ameensol** [comment]
Update:

I realized the two postgres files were the amd64/i386 files. *facepalm* Removing the ppa allowed me to follow through with the update, which I did with `xiwi`. 

Attempting to `sudo startunity` after the update resulted in the following error:

```
Loading extension GLX
Unable to set master
xf86: found device 0
xf86: found device 0
/usr/bin/xinit: XFree86_VT property unexpectedly has 0 items instead of 1
crouton: version 1-20150428153647~master:6ae56c3d
release: trusty
architecture: amd64
xmethod: xiwi
targets: xiwi,unity
host: version 6812.75.2 (Official Build) stable-channel peppy 
kernel: Linux localhost 3.8.11 #1 SMP Tue Apr 14 18:57:27 PDT 2015 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
```

**Comment by ameensol** [comment]
I think `startunity` isn't working because I set the target -t to xiwi instead of unity, so I'm running the update script again with -t unity.

**Comment by ameensol** [comment]
I was wrong. I still get the exact same error output with startunity... looking into GLX

**Comment by ameensol** [comment]
Installed the crouton chrome extension as per the error text:
```
Unable to start display, make sure the crouton extension is installed
and enabled, and up to date. (download from http://goo.gl/OVQOEt)
```

Now, `sudo startunity` works but feels like a shell of what I formerly had. It's terribly slow, the display is unresponsive, and if I want to fullscreen to get rid of that ugly gray top bar I get an "crouton is fullscreen, click to exit" alert message every time I move my mouse to the top 10% of the screen. This is awful! How do I get my fast unity back?

**Comment by ameensol** [comment]
So I deleted my chroot and reinstalled it with only the unity target to avoid having xiwi at all. This worked, but all my files are gone. Installing `gnome-system-monitor` confirms that I have a great deal more free space than I had before... 

Well, this has been a painful process, and is now forever encoded in this issue thread. I hope that there is a better way to get around this segfault issue without inadvertently nuking everything so others don't go through this. Cheers.

**Comment by ameensol** [comment]
Update: 

I let ubuntu upgrade to 14.04.2, after which I ran into the same segfault problem as earlier. Deleting the chroot again and reinstalling fixed it.

**Comment by ameensol** [comment]
Oh. When you use -n it just names it trusty, you have to use -r too to actually install trusty...

**Comment by smuryginim** [comment]
hello! Guys I just install the latest update from Chrome and I face the same issue.
Do someone find a solution to restore ubuntu without reinstall?

**Comment by divx118** [comment]
@smuryginim Did you update your chroot see https://github.com/dnschneid/crouton/wiki/Common-issues-and-reporting

**Comment by smuryginim** [comment]
sudo sh ~/Downloads/crouton -n chrootname -u 
This helped! Thank you very much!

**Comment by Allie15** [comment]
Last year I followed instructions on YouTube for installing and old version of ubuntu (12.04?) I was afraid of something going wrong if I updated it to 14.04 and now it won't work at all.  What do I do?  I know this thread explains what to do, but I don't understand.  Could someone give me step by step instructions?  Or should I powerwash my Chromebook?  Will powerwashing fix everything?

**Comment by divx118** [comment]
@Allie15 Please state what you don't understand about  https://github.com/dnschneid/crouton/wiki/Common-issues-and-reporting if there is something not clear then we can update the wiki.

**Comment by Allie15** [comment]
I'm sure it's my brain that is what needs updating, but eager to learn.

I couldn't get past the first step: Remove any existing installer from your Downloads folder.
How do I do that?

**Comment by divx118** [comment]
@Allie15 
Option 1: open the file app in chromeos and look in Downloads if there is a file crouton, if there is delete it else you are good to proceed. 
Option 2: on the command line you can use `rm -f ~/Downloads/crouton'

**Comment by Allie15** [comment]
Option 1: Huh? where do I look to look in Downloads?

Option 2: In crosh? Do I type "shell" first?

**Comment by divx118** [comment]
Well we go for option 2: yes type shell first

**Comment by Allie15** [comment]
crosh> shell
chronos@localhost / $ rm f ~/Downloads/crouton
rm: cannot remove ‘f’: No such file or directory
rm: cannot remove ‘/home/chronos/user/Downloads/crouton’: No such file or directory
chronos@localhost / $ 

Now what?

**Comment by Allie15** [comment]
crosh> shell
chronos@localhost / $ rm -f ~/Downloads/crouton
chronos@localhost / $ 

Did I do it right?

**Comment by divx118** [comment]
Yes you can continue with the next step on the wiki page.

**Comment by Allie15** [comment]
Fetch a fresh installer: https://goo.gl/fd3zc

I clicked it and it said that the file type isn't supported visit chrome web store

**Comment by divx118** [comment]
If you click the link it should download a file called crouton into your Downloads folder. Once that is done go to the next step in the wiki.

**Comment by Allie15** [comment]
I found crouton in files.  Is it ok that it doesn't give an optioon to open it, but Zip selection (which I don't know how to do)?

Then I tried the sudo edit-chroot -a

crosh> shell
chronos@localhost / $ sudo edit-chroot -a
precise

**Comment by divx118** [comment]
Yes you can't open it in the files app.
Well your chroot is named `precise` So you run in a crosh shell according to the next step in the wiki the following command. `sudo sh ~/Downloads/crouton -n precise -u`

**Comment by Allie15** [comment]
Thank you very much for answering my dummy questions and walking me through it.

It seems to be in working order except my Firestorm Viewer is out of order now.  Do you know anything about that?  That's why I waited so long to update to 14.04 because I wasn't sure if it would work on it.  I thought I should update before trying to familiarize myself a little with programming before taking a computer class.

I can't figure out how to send messages here.

**Comment by radix** [comment]
I'm getting this segfault *sporadically* on an updated Pixel 2 and freshly installed Xenial chroot with `xfce,xiwi` targets. I didn't notice it until I added a `-dpi 239x239` flag to my xiwi's XARGS, but given that it's only happening sporadically that may not be relevant.

**Comment by radix** [comment]
backtrace from my startxiwi command, for reference:

```
(EE) Backtrace:
(EE) 0: /usr/lib/xorg/Xorg (xorg_backtrace+0x4e) [0x7fe8219515ce]
(EE) 1: /usr/lib/xorg/Xorg (0x7fe82179f000+0x1b6959) [0x7fe821955959]
(EE) 2: /lib/x86_64-linux-gnu/libc.so.6 (0x7fe81f42d000+0x354a0) [0x7fe81f4624a0]
(EE) 3: /usr/lib/xorg/Xorg (0x7fe82179f000+0xb2558) [0x7fe821851558]
(EE) 4: /usr/lib/xorg/Xorg (xf86BusProbe+0x9) [0x7fe821828c79]
(EE) 5: /usr/lib/xorg/Xorg (InitOutput+0x774) [0x7fe821837824]
(EE) 6: /usr/lib/xorg/Xorg (0x7fe82179f000+0x57a44) [0x7fe8217f6a44]
(EE) 7: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main+0xf0) [0x7fe81f44d830]
(EE) 8: /usr/lib/xorg/Xorg (_start+0x29) [0x7fe8217e0f59]
(EE) 
(EE) Segmentation fault at address 0x0
(EE) 
Fatal server error:
(EE) Caught signal 11 (Segmentation fault). Server aborting
```
